src/client/tile.cpp

```diff
@@ -1175,7 +1175,18 @@ bool TextureSource::generateImagePart(std::string part_of_name,
 					core::rect<s32>(pos_from, dim),
 					video::SColor(255,255,255,255),
 					NULL);*/
-			blit_with_alpha(image, baseimg, pos_from, pos_to, dim);
+
+			v2u32 dim_dst = baseimg->getDimension();
+			if (dim != dim_dst && dim.Width == dim.Height) {
```

Comment:
Yes, if the check can be removed and it still doesn't break any mods, that would be good since it makes the behaviour of the texture combining code much more predictable (fewer corner cases).